core/azure-core/src/main/java/com/azure/core/implementation/annotation/ServiceClient.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.annotation;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.TYPE;
+
+/**
+ * Annotation given to all service client classes.
+ */
+@Target({TYPE})
+@Retention(RetentionPolicy.SOURCE)
+public @interface ServiceClient {
+
+    /**
+     * The builder class that can construct an instance of this class.
+     * All service clients are instantiated using a builder and this is a required field.
+     * Also, builders should be annotated with {@link ServiceClientBuilder}
+     */
+    Class<?> builder();
+
+    /**
+     * Represents whether the network IO methods on this client will be performed asynchronously or
+     * synchronously (i.e. blocking).
+     */
+    boolean isAsync() default false;
+
+    /**
+     * Optional field to indicate all the services this service client interacts with.
+     * All classes mentioned in this list should be annotated with {@link ServiceInterface}.
+     * Typically, there's one service associated with each client. However, there could be zero to
+     * N services associated with a single client.
+     * @return An array of all services this service client interacts with
```

Comment:
I'm not sure we need to support zero to N, zero to one is probably sufficient. I also wonder if we should name this `serviceInterface` or something?

Reply:
The reason I made this an array was because of the spec we had here - https://jogiles.z19.web.core.windows.net/java-spec-draft/JavaSpec.html#general-terminology

Client Library section:
`There is generally a client library per Azure service and per target language. Sometimes a single client library will contain the ability to connect to multiple services.`

Reply:
I agree with your argument. We can leave it as an array.